x-pack/plugins/security_solution/server/endpoint/utils/create_internal_readonly_so_client.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest, SavedObjectsClientContract, SavedObjectsServiceStart } from 'kibana/server';
+import { EndpointError } from '../errors';
+
+type SavedObjectsClientContractKeys = keyof SavedObjectsClientContract;
+
+const RESTRICTED_METHODS: readonly SavedObjectsClientContractKeys[] = [
```

Comment:
I initially wanted to do "allow" list rather than "deny" list, but found that the SO client attempts to access other internal properties on the client that are not actually exposed via the `SavedObjectsClientContract`. Some of the methods below I don't even know what they do (`openPointINTimeForType`??), but since they sounded like to me that they might possibly be mutating type of methods, I block them.